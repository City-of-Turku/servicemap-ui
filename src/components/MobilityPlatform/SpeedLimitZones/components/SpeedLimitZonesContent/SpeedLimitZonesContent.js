import React from 'react';
import { PropTypes } from 'prop-types';
import { Typography } from '@mui/material';
import { useIntl } from 'react-intl';
import { StyledContainer, StyledHeaderContainer, StyledTextContainer } from '../../../styled/styled';

const SpeedLimitZonesContent = ({ item }) => {
  const intl = useIntl();

  return (
    <StyledContainer>
      <StyledHeaderContainer>
        <Typography variant="subtitle1" component="h3">
          {intl.formatMessage({
            id: 'mobilityPlatform.content.speedLimitZones.area',
          })}
        </Typography>
      </StyledHeaderContainer>
      <div>
        <StyledTextContainer>
          <Typography>
            {intl.formatMessage(
              {
                id: 'mobilityPlatform.content.speedLimitZones.limit',
              },
              { item: item.extra.speed_limit },
            )}
          </Typography>
        </StyledTextContainer>
      </div>
    </StyledContainer>
  );
};

SpeedLimitZonesContent.propTypes = {
  item: PropTypes.shape({
    extra: PropTypes.objectOf(PropTypes.number),
  }).isRequired,
};

export default SpeedLimitZonesContent;
