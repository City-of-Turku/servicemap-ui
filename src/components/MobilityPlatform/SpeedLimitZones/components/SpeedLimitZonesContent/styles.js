export default theme => ({
  subtitle: {
    marginBottom: theme.spacing(1),
    paddingBottom: theme.spacing(0.5),
    borderBottom: '1px solid #000000',
    width: '88%',
  },
});
