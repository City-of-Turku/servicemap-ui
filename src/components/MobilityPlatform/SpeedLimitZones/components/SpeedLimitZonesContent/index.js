import SpeedLimitZonesContent from './SpeedLimitZonesContent';

export default SpeedLimitZonesContent;
