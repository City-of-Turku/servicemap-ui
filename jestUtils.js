import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });