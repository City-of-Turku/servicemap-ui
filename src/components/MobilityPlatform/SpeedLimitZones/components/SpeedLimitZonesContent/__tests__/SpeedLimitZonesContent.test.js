// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../i18n/fi';
import SpeedLimitZonesContent from '../index';

const mockProps = {
  item: {
    extra: {
      speed_limit: 40,
    },
  },
};

const renderWithProviders = getRenderWithProviders({});

describe('<SpeedLimitZonesContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<SpeedLimitZonesContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<SpeedLimitZonesContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    const h6 = container.querySelector('h6');
    expect(h6.textContent).toContain(finnishTranslations['mobilityPlatform.content.speedLimitZones.area']);
    expect(p[0].textContent).toContain('Nopeusrajoitus: 40 km/t');
  });
});
