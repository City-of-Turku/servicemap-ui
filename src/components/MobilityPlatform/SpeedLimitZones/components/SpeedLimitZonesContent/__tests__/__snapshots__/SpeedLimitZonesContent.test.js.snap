// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpeedLimitZonesContent /> should match snapshot 1`] = `
<div>
  <div
    class="injectIntl(SpeedLimitZonesContent)-padding-1"
  >
    <div
      class="injectIntl(SpeedLimitZonesContent)-subtitle-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Nopeusrajoitusalue
      </h6>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Nopeusrajoitus: 40 km/t
    </p>
  </div>
</div>
`;
